app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -40,21 +40,45 @@ public static void performNewBrowserSessionCleanup() {
         // Nothing: a WebKit work-around.
     }
 
-    public static class GeckoWebView extends NestedGeckoView implements IWebView {
+    public static class GeckoWebView extends NestedGeckoView implements IWebView, SharedPreferences.OnSharedPreferenceChangeListener {
```

Comment:
\[PMD\] INFO: The class 'GeckoWebView' has a Standard Cyclomatic Complexity of 2 \(Highest = 10\)\.